pkg/controller/storageprofile-controller.go

```diff
@@ -296,13 +298,26 @@ func (r *StorageProfileReconciler) computeMetrics(profile *cdiv1.StorageProfile,
 		return err
 	}
 
+	isSNO := false
+	clusterInfra := &ocpconfigv1.Infrastructure{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, clusterInfra); err != nil {
+		if !meta.IsNoMatchError(err) && !k8serrors.IsNotFound(err) {
+			return err
+		}
+	} else {
+		isSNO = clusterInfra.Status.ControlPlaneTopology == ocpconfigv1.SingleReplicaTopologyMode &&
+			clusterInfra.Status.InfrastructureTopology == ocpconfigv1.SingleReplicaTopologyMode
+	}
```

Comment:
So with LVMS this check was quite tricky to rely on, since you could be doing "SNO" with multiple workers but your images (and thus VMs) would end up on same node, so for our purposes it wasn't "SNO".

Do we have the same concern here? I want to make sure we don't take away the alert from cases where it's actually helpful

Reply:
cc @awels wdyt

Reply:
Discussing it with @aglitke , we currently decide to suppress only in the case of a single node, so afaiu in the mentioned LVMS case we will still alert if no RWX.

Reply:
The problem is (I think) that this API does not actually guarantee 1 single node.
So someone doing 3-node SNO setup, trying to properly set up live migration, will not receive the alert
We may or may not be okay with that - but something to consider

Reply:
The [doc](https://docs.openshift.com/container-platform/4.15/operators/operator_sdk/osdk-ha-sno.html) says that's the way to detect a single-node, and afaik the other alternative for detecting SNO requires adding cluster rbac for node “list” and “watch”, which we currently prefer not to add.

Reply:
Yeah as I thought they use the word "single" but it doesn't actually guarantee a single node:

- When the control plane replica count is less than 3, the controlPlaneTopology status is set to SingleReplica. Otherwise, it is set to HighlyAvailable.

- When the worker replica count is 0, the control plane nodes are also configured as workers. Therefore, the infrastructureTopology status will be the same as the controlPlaneTopology status.

Reply:
Again I am not strictly against just not firing the alert for anything that's not HA, but have to raise this point

Reply:
btw why is it so *bad* to add cluster rbac for node “list” and “watch”? Sure CDI is a completely different animal, but HCO has this rbac.

Reply:
When adding RBAC we always have to consider why are we adding RBAC. And honestly some alert doesn't seem like the right thing to add RBAC for.

Reply:
@awels understood. And what about @akalenyu argument here?

Reply:
I think this was @mhenriks point, that it is not easy to get right to figure out if we are truly in a single node environment or not. There are all kinds of edge cases where we won't alert with this and probably should. I do still think it is an improvement over always alerting in the wrong case.

Reply:
That is another reason why adding the nodes is not a good idea. You could have multiple nodes, and some of them are not marked as worker nodes, and you now have only one node that can run workloads. Getting that stuff right is very tricky. I think this is 'good enough'. People with those edge cases should know what they are doing.